Firmware/Marlin_main.cpp

```diff
@@ -9512,16 +9512,8 @@ void ThermalStop(bool allow_recovery)
 
                 // we cannot make a distinction for the host here, the pause must be instantaneous
                 // so we call the lcd_pause_print to save the print state internally. Thermal errors
-                // disable heaters and save the original temperatures to saved_*, which will get
-                // overwritten by stop_and_save_print_to_ram. For this corner-case, re-instate the
-                // original values after the pause handler is called.
-                uint8_t bed_temp = saved_bed_temperature;
-                uint16_t ext_temp = saved_extruder_temperature;
-                uint8_t fan_speed = saved_fan_speed;
+                // disable heaters and save the original temperatures to saved_*
                 lcd_pause_print();
-                saved_bed_temperature = bed_temp;
-                saved_extruder_temperature = ext_temp;
-                saved_fan_speed = fan_speed;
             }
```

Comment:
Doesn't think break exactly the scenario written in the comment?

A thermal error will save temperatures into saved_* (see set_temp_error), then disable heaters and call ThermalStop. Calling lcd_pause_print will then save "0" into saved_, so resuming from a thermal error will restore broken temperatures?

Reply:
@gudnimg ???

Reply:
@wavexx Thanks for looking into this

Reply:
@wavexx @3d-gussner 

I don't see how "0" is saved with these changes. This is how I see the code running:

First `set_temp_error()` is called and it calls `refresh_print_state_in_ram();` This will set `isPartialBackupAvailable = true`.

Then `lcd_pause_print();` is executed and calls `stop_and_save_print_to_ram()`. Since `isPartialBackupAvailable = true`, we do not refresh the print state in RAM (i.e. overwrite `saved_extruder_temperature` since we already have a valid partial backup).

The "partial backup" was added very recently to support Power Panic when the extruder is parked for example in M600, Paused print, MMU error etc.

Does this make sense?

Reply:
Am I looking at the wrong piece of code maybe?

https://github.com/prusa3d/Prusa-Firmware/blob/ff42510a5c9a97601e6a912495694796f9d053ae/Firmware/temperature.cpp#L511

there's no call to refresh_print_state_in_ram(). It sets saved_, then disables the heater.

Reply:
Duh.. looks like I was. I didn't realize set_temp_error was changed within the PR itself.

Reply:
Re-looking at the new lcd pause and set_temp_error code, my point is moot. I don't see a problem with this. It's definitely much better compared to what we had before where we would always overwrite saved_ values.